### ISSUE TITLE
Multilanguage Support

### ISSUE THREAD

**Issue by Joly0** [fmasa/wfrp-master#24]
Hey, i have been using your app and it works great so far, though to recommend this app to friends of mine, it would be great, if there was multilanguage support. 
My friends arent that familiar with english, so it would be hard for them to use this otherwise great tool.
If contributors could add their own languages to this app, it would be a great addition.

**Comment by fmasa** [comment]
Hi,
there are two options IMO.

#### "Official" translation in app

##### Pros

- great from translation coverage perspective (we can make sure that all texts have translations before release)

##### Cons

- This would require native speaker for each of the supported languages available to update the translations before the release.

#### Translations import

##### Pros

- Fairly simple to implement from my side, everything Is already spe

##### Cons

- new/changed texts would break existing translations

---

Right now I'm not sure I want to support the first option. The second one I'm not sure.

Do you have a particular language in mind?

**Comment by Joly0** [comment]
My idea on this was about using generic placeholders in the app for text fields and let contributors add new languages through resource files and if a field is not yet translated, then default to english.
How the translated files get loaded in the app could either be at build time so in the released apk is the latest language changes included. Otherwise it should be possible to dynamically load the languages on the go.

Though this is just my view on this. In my case the needed language would be german and i would be happy to help translate the app later, when this feature is implemented

**Comment by fmasa** [comment]
@Joly0 Most of the values is currently externalized in [Strings class](https://github.com/fmasa/wfrp-master/blob/88fa7da27342c431fd0b8980d010be1bbb711306/common/src/commonMain/kotlin/cz/frantisekmasa/wfrp_master/common/localization/Strings.kt) as string placeholders. There are a few lambdas that I'll rewrite to string placeholders as well. Then I'll think about some injection points for the translations.

**Comment by Joly0** [comment]
Hey, any news on this one? :D

**Comment by fmasa** [comment]
@Joly0, @Stilzkin I am moving the translations to separate resource files for more straightforward translations and I will also set up a [Weblate](https://weblate.org) project so that anyone can contribute the translations. I expect this to be ready around this weekend.

**Comment by Stilzkin** [comment]
Hi! I've completed the Italian translation of Strings.kt (4 hr well spent :D) but i dont know how to implement in the project (im fresh new to github).

[Italian.zip](https://github.com/fmasa/wfrp-master/files/12322973/Italian.zip)

**Comment by Joly0** [comment]
> @Joly0, @Stilzkin I am moving the translations to separate resource files for more straightforward translations and I will also set up a [Weblate](https://weblate.org) project so that anyone can contribute the translations. I expect this to be ready around this weekend.

Sounds great, i am already hyped for it and ready for translating :D

**Comment by fmasa** [comment]
> Hi! I've completed the Italian translation of Strings.kt (4 hr well spent :D) but i dont know how to implement in the project (im fresh new to github).

Great, thank you! I'll add it on my side once I'm done with the some necessary changes. I'll also write some basic instructions for translation submissions.  :slightly_smiling_face: 

> Sounds great, i am already hyped for it and ready for translating :D

I'm looking forward to it! :slightly_smiling_face:

**Comment by fmasa** [comment]
@Stilzkin I processed everything and set up Weblate as a tool that would simplify contributions. You can see that there are still a few translations missing in Italian (https://hosted.weblate.org/projects/wfrp-master/glossary/). Also, I encountered some inconsistencies (NPCS are sometimes called `PNG`, and sometimes `NPC`).

You can submit the changes directly there and the Weblate should notify me, so I can automatically apply these translations.

Also please note, that changes hitting this GitHub repository (master branch more specifically) are not immediately available in the app. The changes become available only after I tag a new release. I'll document this in more detail in a few days.

---

@Joly0  You can take a look at [Weblate](https://hosted.weblate.org/projects/wfrp-master/glossary/) now as it should be possible to add a new language :slightly_smiling_face: Alternatively if you are familiar with git, the https://github.com/fmasa/wfrp-master/commit/957f6727b322b993da492a205984d581855a8ef0 can work as inspiration for direct contribution as PR.

**Comment by Stilzkin** [comment]
> @Stilzkin I processed everything and set up Weblate as a tool that would simplify contributions. You can see that there are still a few translations missing in Italian (https://hosted.weblate.org/projects/wfrp-master/glossary/). Also, I encountered some inconsistencies (NPCS are sometimes called `PNG`, and sometimes `NPC`), would you mind taking a look? 🙂
> 
> You can submit the changes directly there and the Weblate should notify me, so I can automatically apply these translations.
> 
> Also please note, that changes hitting this GitHub repository (master branch more specifically) are not immediately available in the app. The changes become available only after I tag a new release. I'll document this in more detail in a few days.
> 
> @Joly0 You can take a look at [Weblate](https://hosted.weblate.org/projects/wfrp-master/glossary/) now as it should be possible to add a new language 🙂 Alternatively if you are familiar with git, the [957f672](https://github.com/fmasa/wfrp-master/commit/957f6727b322b993da492a205984d581855a8ef0) can work as inspiration for direct contribution as PR.

Nice work @fmasa ! I look for corrections asap!

**Comment by Stilzkin** [comment]
OK, I translated the remaining strings and fixed a few things (including NPC / PNG). One question: where are the names of the Conditions? I didn't see them in the file.

**Comment by fmasa** [comment]
I see now that Conditions are not translatable, I'll fix that.

**Comment by fmasa** [comment]
@Stilzkin I accepted your translations https://github.com/fmasa/wfrp-master/pull/152.

I also merged https://github.com/fmasa/wfrp-master/pull/151, so Conditions can now be translated.

**Comment by Stilzkin** [comment]
> @Stilzkin I accepted your translations #152.
> 
> I also merged #151, so Conditions can now be translated.

Conditions translated 😉

**Comment by fmasa** [comment]
@Stilzkin Would you take a look once more? There are additional fields to translate because I added Trapping compendiums (a way to import Trappings from CRB, Winds of Magic, and Up in Arms).

As mentioned, this is not a problem as any missing translation will fall back into English. But I think it would be awesome if the first release with Italian support would have 100% translation. I don't plan any other significant changes for 3.14.0, so don't worry that I would be asking this every week :grinning: 

Also, in the future, there will almost certainly be changes in UI that will bring new strings that could use a translation. AFAIK Weblate supports notifications when new strings are available.

**Comment by fmasa** [comment]
Closing this issue, as it's done from my side. 
Anyone who wants to contribute can read the instructions at https://github.com/fmasa/wfrp-master/wiki/Translations

@Joly0 as mentioned in WIKI, just open a new issue once you finish the German translation and I'll add it to the next version :slightly_smiling_face:
